### PR TITLE
Fix SSA for APIService

### LINKED ISSUE #89264: Server-side apply doesn't work for APIService
It looks like the openapi is not routed properly to set-up the apiservice so apply requests will fail because the fieldmanager is not set-up properly.

/cc @jennybuckley @liggitt 
/sig api-machinery
/priority important-soon
/kind bug

### PR DESCRIPTION
**Pull request by kevindelgado**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
It removes the code to nil out the OpenAPIConfig on the aggregator server so that SSA requests are routed properly for the apiservices resource.

**Which issue(s) this PR fixes**:

Fixes #89264

```release-note
Fixes using server-side apply with APIService resources
```

### CODE REVIEW

File: staging/src/k8s.io/api/core/v1/types.go [hunk 1]
```diff
@@ -4265,6 +4265,7 @@ type ServicePort struct {
 
 	// The IP protocol for this port. Supports "TCP", "UDP", and "SCTP".
 	// Default is TCP.
+	// +default="TCP"
```
Comment (apelisse) [inline staging/src/k8s.io/api/core/v1/types.go:4268]: @liggitt Should we back-port that in 1.20? If so, we should maybe make that into a separate PR.

**Comment by apelisse** [comment]
LGTM

File: staging/src/k8s.io/api/core/v1/types.go (hunk 1 above)
Comment (liggitt) [reply staging/src/k8s.io/api/core/v1/types.go:4268]: It's in a separate commit we can pick if needed. If it's not a regression in 1.20 and we don't have calls for it in bugs, I'd probably just let it ride in 1.21

**Comment by kevindelgado** [comment]
I had to update generated proto and couldn't do it on Jordan's branch. New PR is at #98576
